ivy_tests/test_ivy/helpers.py

```diff
@@ -1803,37 +1807,46 @@ def array_values(
     values = None
     if "int" in dtype:
```

Comment:
we should actually include the `small_value_safety_factor` for integer types as well, with `1` then being the smallest magnitude value generated, for any case where `small_value_safety_factor < 1`. This will enable us to prevent zero divisions via the safety factor when generating integers, as an integer divided by an integer leads to a float.

Reply:
@djl11 
Would it be right to do it for `int8`, `int16` etc. because this would result in negative values not being generated?

Reply:
you should do something like:

```
min_value = ivy.default(min_value, round(-128 * large_value_safety_factor))
max_neg_value = -1 if small_value_safety_factor > 1 else 0
min_pos_value = 1 if small_value_safety_factor > 1 else 0
max_value = ivy.default(max_value, round(127 * large_value_safety_factor))
```

and then at the end:

```
values = draw(list_of_length(x=st.integers(min_value, max_neg_value) | st.integers(min_pos_value, max_value), length=size))
```